server/user.go

```diff
@@ -295,18 +313,30 @@ func (s *UserMgmtServer) getCreds(r *http.Request, requiresAdmin bool) (api.Cred
 		return api.Creds{}, err
 	}
 
-	isAdmin, err := s.cm.IsDexAdmin(clientID)
-	if err != nil {
-		log.Errorf("userMgmtServer: GetCreds err: %q", err)
-		return api.Creds{}, err
+	i := 0
+	for _, clientID := range clientIDs {
+		// Make sure the client actually exists.
+		isAdmin, err := s.cm.IsDexAdmin(clientID)
+		if err != nil {
+			log.Errorf("userMgmtServer: GetCreds err: %q", err)
+			return api.Creds{}, err
+		}
+		// If the endpoint requires an admin client, filter out clients which are not admins.
+		if isAdmin || !requiresAdmin {
+			clientIDs[i] = clientID
+			i++
+		}
 	}
-	if requiresAdmin && !isAdmin {
+
+	clientIDs = clientIDs[:i]
```

Comment:
After checking `ok` (line 278), shouldn't we reduce clientIDs to the set of client IDs that verify successfully? Otherwise, I can imagine a case where several clients who claim to be admins but do not verify, and one verified client who is not an admin, would pass the `ok` check; then then the bad admins would get filtered into this new splice even though they do not verify.